src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -294,7 +294,7 @@ static int wall_tty_block(void) {
         (void) mkdir_parents_label(p, 0700);
         (void) mkfifo(p, 0600);
 
-        fd = open(p, O_RDONLY|O_CLOEXEC|O_NONBLOCK|O_NOCTTY);
+        fd = open(p, O_PATH|O_CLOEXEC|O_NONBLOCK|O_NOCTTY);
```

Comment:
this defeats the point of the fifo we have here. See the comment in wall_tty_match()

Reply:
Ok, so that open is only needed to check if the other end of the pipe is valid?
Dropping.